src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+
+  let currentState = { ...state };
```

Comment:
[NAMING] - The variable name `currentState` is a bit misleading because it suggests that it will be mutated, which is against the task's requirements. Consider renaming it to something like `stateCopy` to reflect that it is a copy of the state that can be safely modified.